spec/models/organization_spec.rb

```diff
@@ -138,11 +139,43 @@
     end
   end
 
-  describe 'Callbacks' do
-    it 'generates the api key' do
-      organization.save!
+  describe '#save' do
+    subject { organization.save! }
 
-      expect(organization.api_key).to be_present
+    before do
+      allow(organization).to receive(:generate_document_number_prefix).and_call_original # rubocop:disable RSpec/SubjectStub
```

Comment:
Stubbing private method is not a good idea (that's why the rubocop rule is here :) )
I think it's better to check that after a save, the expected values are set on the model.

The point is that most of the time specs should not be based on the inner implementation, but on the expected result otherwise the test might be passing even if the what is tested is not working as expected

Reply:
Regarding rubocop, there're no warning about stubbing that private method. Maybe we need to check our Rubocop config?

The disabled cop `RSpec/SubjectStub` has a bit different purpose – it ensures that we do not stub thing that we're testing. 
In this case it has false positive match since subject in my case is different, testing that method is called by when another action is done allowed behaviour.